families-api/app/main.py

```diff
@@ -0,0 +1,225 @@
+from typing import Generic, Optional, TypeVar
+
+from fastapi import APIRouter, Depends, FastAPI
+from pydantic_settings import BaseSettings
+from sqlalchemy import create_engine
+from sqlmodel import Field, Relationship, Session, SQLModel, func, select
+
+
+class FamilyCorpusLink(SQLModel, table=True):
+    __tablename__ = "family_corpus"  # type: ignore[assignment]
+    corpus_import_id: int | None = Field(
+        default=None, foreign_key="corpus.import_id", primary_key=True
+    )
+    family_import_id: int | None = Field(
+        default=None, foreign_key="family.import_id", primary_key=True
+    )
+
+
+class Corpus(SQLModel, table=True):
+    __tablename__ = "corpus"  # type: ignore[assignment]
+    import_id: str | None = Field(default=None, primary_key=True)
```

Comment:
question: what behaviour does giving a default=None entail? And what's the logic behind nullable primary key? That feels dangerous to me

Reply:
I've made all IDs required (copy/pasta from elsewhere)